README.md
# TWITTER: WHAT’S HAPPENING???

[![Release](https://img.shields.io/github/v/release/ttldtor/WHAT-S-HAPPENING-)](https://github.com/ttldtor/WHAT-S-HAPPENING-/releases/latest)
[![License](https://img.shields.io/badge/license-Unlicense-orange)](https://github.com/ttldtor/WHAT-S-HAPPENING-/blob/master/LICENSE)

## Examples

![Example1](https://user-images.githubusercontent.com/3264871/177410388-ab8a83b7-3fa7-4cd2-8f94-c166e23349ce.png)
![Example2](https://user-images.githubusercontent.com/3264871/177218610-01dc80f3-db33-4010-9cd3-d41ff569c69b.png)

BTC: bc1q779qrrmy22aqtvc00khnmefjuycprg292smpac


ReleaseNotes.md
Release Notes
==

v3.6.0

- ПОШУК =3

v3.5.0

- 🎁 Читаю -> УЖЕ ПОДПИЩЩЕК!
- 🍔 Перестать читать -> ДОСТАТОЧНО!
- 🍗 Начать продвижение -> СДЕЛАТЬ ПОТРАЧЕНО!

v3.4.0

- 🎁 Читать -> СТАТЬ ПОДПИЩЩЕКОМ!

v3.3.0

- 🎁 Changed the description
- ➕ Added a CI\CD workflow

v3.2.0

- 🤣 Added some badges
- 🤣 Added the translation for 'Твитнуть все'

v3.1.0

- 😉 More translations
- 😉 The selectors were simplified

v3.0.0

- ❤ Added `@include`
- 🍕 userscript.js renamed to whatshappening.user.js

v2.0.0

- 🎉 Added the RU translation for the MAIN twitter \[button\]: Твитнуть -> ХУЙНУТЬ
- 💌 Added an example to the README

v1.0.2

- ❤ Added the `@updateURL`, `@downloadURL` & `@hompageURL`

v1.0.1

- 🔧 Fixed the README.md file (the project "description")

v1.0.0

- 😀 The init commit: added some translations (RU, EN, DE, FR) for the Draft editor placeholder


userscript.js
// ==UserScript==
// @name        Twitter: WHAT’S HAPPENING???
// @name:ru     Twitter: А ЧТО ПРОИСХОДИТ???
// @name:de     Twitter: WAS IST LOS???
// @name:fr     Twitter: QUE SE PASSE-T-IL???
// @namespace   https://github.com/ttldtor
// @compatible  chrome
// @compatible  firefox
// @match       *://*twitter.com/*
// @include     *://*twitter.com/*
// @grant       none
// @version     2.0.0
// @author      ttldtor
// @description kek
// @icon        https://www.google.com/s2/favicons?sz=64&domain=twitter.com
// @license     Unlicense
// @homepageURL https://github.com/ttldtor/WHAT-S-HAPPENING-
// @downloadURL https://raw.githubusercontent.com/ttldtor/WHAT-S-HAPPENING-/default/userscript.js
// @updateURL   https://raw.githubusercontent.com/ttldtor/WHAT-S-HAPPENING-/default/userscript.js
// ==/UserScript==

(function(){
  'use strict';
  
  const translations = new Map();
  
  translations.set('Что происходит?', 'А ЧТО ПРОИСХОДИТ???');  
  translations.set('What’s happening?', 'WHAT’S HAPPENING???');
  translations.set('Was gibt\'s Neues?', 'WAS IST LOS???');
  translations.set('Quoi de neuf ?', 'QUE SE PASSE-T-IL???');
  
  const buttonsTranslations = new Map();
  
  buttonsTranslations.set('Твитнуть', 'ХУЙНУТЬ');
  
  function update() {
    Array.from(document.querySelectorAll('div.public-DraftEditorPlaceholder-inner'))
      .forEach(draftEditorPlaceholder => {
        const translation = translations.get(draftEditorPlaceholder.textContent);

        if (translation !== undefined) {
          draftEditorPlaceholder.textContent = translation;
        }
      });
    
    Array.from(document.querySelectorAll('div[role="button"] span span'))
      .forEach(buttonTextSpan => {
        const translation = buttonsTranslations.get(buttonTextSpan.innerHTML);

        if (translation !== undefined) {
          buttonTextSpan.innerHTML = translation;
        }
      });
  }
  
  setInterval(update, 500);  
})();


whatshappening.user.js
// ==UserScript==
// @name        Twitter: WHAT’S HAPPENING???
// @name:ru     Twitter: А ЧТО ПРОИСХОДИТ???
// @name:de     Twitter: WAS IST LOS???
// @name:fr     Twitter: QUE SE PASSE-T-IL???
// @namespace   https://github.com/ttldtor
// @compatible  chrome
// @compatible  firefox
// @match       *://*twitter.com/*
// @include     *://*twitter.com/*
// @grant       none
// @version     3.6.0
// @author      ttldtor
// @description This extension changes placeholders on edit fields, as well as the text on some buttons.
// @icon        https://www.google.com/s2/favicons?sz=64&domain=twitter.com
// @license     Unlicense
// @homepageURL https://github.com/ttldtor/WHAT-S-HAPPENING-
// @downloadURL https://raw.githubusercontent.com/ttldtor/WHAT-S-HAPPENING-/default/whatshappening.user.js
// @updateURL   https://raw.githubusercontent.com/ttldtor/WHAT-S-HAPPENING-/default/whatshappening.user.js
// ==/UserScript==

(function(){
  'use strict';
  
  const translations = new Map();
  
  translations.set('Что происходит?', 'А ЧТО ПРОИСХОДИТ???');  
  translations.set('What’s happening?', 'WHAT’S HAPPENING???');
  translations.set('Was gibt\'s Neues?', 'WAS IST LOS???');
  translations.set('Quoi de neuf ?', 'QUE SE PASSE-T-IL???');
  translations.set('Поиск в твиттере', 'ПОШУК!');
  
  const buttonsTranslations = new Map();
  
  buttonsTranslations.set('Твитнуть', 'ХУЙНУТЬ');
  buttonsTranslations.set('Ответить', 'ХУЙНУТЬ В ОТВЕТ');
  buttonsTranslations.set('Твитнуть в ответ', 'ХУЙНУТЬ В ОТВЕТ');
  buttonsTranslations.set('Твитнуть все', 'ХУЙНУТЬ ВСЁ');
  buttonsTranslations.set('Добавить еще один твит', 'ХУЙНУТЬ ЕЩЁ');
  buttonsTranslations.set('Читать', 'СТАТЬ ПОДПИЩЩЕКОМ!');
  buttonsTranslations.set('Читаю', 'УЖЕ ПОДПИЩЩЕК!');
  buttonsTranslations.set('Перестать читать', 'ДОСТАТОЧНО!');
  buttonsTranslations.set('Начать продвижение', 'СДЕЛАТЬ ПОТРАЧЕНО!');

  function findTranslation(translations, text) {
    return translations.reduce((a, b) => {
      if (a !== undefined) {
        return a;
      }

      return b.get(text);
    }, undefined);
  }

  function applyTranslation(element, translation) {
    if (translation === undefined) {
      return;
    }

    if (element.innerHTML !== undefined) {
      element.innerHTML = translation;
    } else if (element.textContent !== undefined) {
      element.textContent = translation;
    }
  }
  
  function update() {
    Array.from(document.querySelectorAll('div.public-DraftEditorPlaceholder-inner'))
      .forEach(draftEditorPlaceholder => {
        applyTranslation(draftEditorPlaceholder, findTranslation([translations, buttonsTranslations], draftEditorPlaceholder.textContent));
      });
    
    Array.from(document.querySelectorAll('div[role="button"] span, a[role="link"] span, a[role="presentation"] span'))
      .forEach(buttonTextSpan => {
        applyTranslation(buttonTextSpan, findTranslation([buttonsTranslations], buttonTextSpan.innerHTML));
      });
  }
  
  setInterval(update, 500);  
})();
